src/components/defaults.js

```diff
@@ -1,7 +1,13 @@
-import { checkHasInstalledExtension } from '../CozyUtils'
-
-export const checkShouldUnlock = async (vaultClient, client) => {
-  return (
-    (await checkHasInstalledExtension(client)) && (await vaultClient.isLocked())
-  )
+/**
+ * Check whether or not the vault should be unlocked
+ *
+ * @param {object} vaultClient - The vault client
+ * @param {function} onBeforeShouldUnlock - An additional check method to unlock
+ * @returns {boolean} True if the vault should be unlocked
+ */
+export const checkShouldUnlock = async (vaultClient, onBeforeShouldUnlock) => {
+  if (onBeforeShouldUnlock) {
+    return (await onBeforeShouldUnlock()) && (await vaultClient.isLocked())
+  }
+  return vaultClient.isLocked()
```

Comment:
:recycle: by using a default param we can simplify this block of code

Example (not tested):
```suggestion
export const checkShouldUnlock = async (
  vaultClient,
  onBeforeShouldUnlock = () => true
) => {
  return (await onBeforeShouldUnlock()) && (await vaultClient.isLocked())
}
```